core/server/common/src/main/java/alluxio/master/journal/AsyncJournalWriter.java

```diff
@@ -46,10 +82,21 @@
   private final long mFlushBatchTimeNs;
 
   /**
-   * Use a {@link ReentrantLock} to guard the journal writing. Using the fairness policy seems to
-   * result in better throughput.
+   * List of flush tickets submitted by ::flush() method.
    */
-  private final ReentrantLock mFlushLock = new ReentrantLock(true);
+  @GuardedBy("mTicketLock")
+  private final List<FlushTicket> mTicketList = new ArrayList<>(200);
```

Comment:
would it make sense to use a concurrent list, and not use an external lock?

Reply:
That could cause flush thread to miss some tickets as concurrent lists don't always provide the final view of the list.

Reply:
I think we can consider this "list" as a producer/consumer queue. The clients add flush requests, and the flush thread notifies them. In the newest iteration of the code, the flush thread will always periodically wake up to write entries, and to notify tickets, so in that sense, we don't need a lock on the ticket queue.

Even the signal from client to flush thread doesn't seem necessary anymore, since the flush thread is always waking up periodically. However, I don't know how that will impact the latency.